src/java.desktop/share/classes/java/awt/image/ComponentSampleModel.java

```diff
@@ -1224,8 +1223,6 @@ public int hashCode() {
             hash ^= bankIndices[i];
             hash <<= 8;
         }
-        hash ^= numBands;
-        hash <<= 8;
```

Comment:
Since this mistake was done in two places I think this is a typo here and not an intentional thing.

Reply:
I am a bit confused by this comment :) I suspect that `equals` got duplicated `numBands` first, and then whoever did the `hashcode` just hashed the same fields in the same order, duplicating it again.

Reply:
I hope so, and not in the opposite order. I'll try to check the history f this file.

Reply:
Please do. I'll wait for a bit.

Reply:
I have found this bug, and the fix for it just implemented both methods at once, so this is a typo.
https://bugs.openjdk.java.net/browse/JDK-4430788

Reply:
Brilliant, thanks. I'll integrate now.